### DISCUSSION TITLE
How to import and use NextUI components in a react app?

### DISCUSSION THREAD

**Discussion by Samsonroyal** [heroui-inc/heroui#188] [Q&A] [answered]
Hello dear NextUI devs;

I have tried using the NextUI components in a demo react app; but I keep receiving errors; I don't know if I am doing it right. I will need some clarification. 

###  The error log is displayed here

![ss1](https://user-images.githubusercontent.com/26835888/151670002-3bbc625c-1221-4974-b053-eaf4b2d6d028.png)

````
### Code from my app.js
`import` { NextUIProvider } from '@nextui-org/react';
import { ReactComponent as Logo } from "./logo.svg";
import Posts from "./components/Posts";
import Layout from "./components/Layout";
import './App.css';
import * as React from 'react';
function App() {
  return (
    <div className="App">
      <NextUIProvider>
        <Layout>
      </NextUIProvider>
      <header className="App-header">
      <Logo style={{ height: 200 }} />  
        <h1>Samson Galactic Posts</h1>
      </header>
      <Layout/>
      <Posts />
    </div>
  );
}
export default App;

````
### Code from Layout.js (a component I am importing)

`import { Grid } from '@nextui-org/react';
    const Layout() => {
      return (
          <ExampleBlock height={100}>
              <Text h6 size={15} color="white" style={{ margin: 0 }}>
                  {text}
              </Text>
          </ExampleBlock>
      );
  }
  return (
  <Grid.Container gap={2} justify="center">
    <Grid xs={4}>
      <Layout text="1 of 3" />
    </Grid>
    <Grid xs={4}>
      <Layout text="2 of 3" />
    </Grid>
    <Grid xs={4}>
      <Layout text="3 of 3" />
    </Grid>
  </Grid.Container>
  ) }
  export default Layout;`

**Comment by jrgarciadev** [comment, marked as answer]
Hey @Samsonroyal it's because you are not closing the first `Layout` tag, try doing it this way:

```jsx
import * as React from 'react';
import { NextUIProvider } from '@nextui-org/react';
import { ReactComponent as Logo } from "./logo.svg";
import Posts from "./components/Posts";
import Layout from "./components/Layout";
import './App.css';

function App() {
  return (
    <div className="App">
      <NextUIProvider>
        <Layout />
      </NextUIProvider>
      <header className="App-header">
      <Logo style={{ height: 200 }} />  
        <h1>Samson Galactic Posts</h1>
      </header>
      <Layout />
      <Posts />
    </div>
  );
}
export default App;
```

I hope it helps you 👍🏻

**Reply by Samsonroyal** [reply]
Hello @jrgarciadev thanks for sparing time to answer my question; I applied the changes but I have more errors; seems something is wrong with my layout.js file. This is the output in my terminal

![error1](https://user-images.githubusercontent.com/26835888/151673824-e7139a53-fb87-418a-af0c-b0895b291e66.png)
`

**Reply by etherealsunshine** [reply]
you need to specify an initial value for the constant with the = sign
